test/gtest/common/mem_buffer.cc

```diff
@@ -302,6 +302,94 @@ void mem_buffer::copy_from(void *dst, const void *src, size_t length,
     }
 }
 
+void mem_buffer::copy_between(void *dst, const void *src, size_t length,
+                              ucs_memory_type_t dst_mem_type,
+                              ucs_memory_type_t src_mem_type)
+{
+    void UCS_V_UNUSED *tmp;
```

Comment:
1. use cudaMemcpyDefault/hipMemcpyDefault to unite conditions:


if (HOST_ACCESSIBLE(src) && HOST_ACCESSIBLE(dst)) { memcpy }
else if (CUDA_ACCESSIBLE(src) && CUDA_ACCESSIBLE(dst)) { cudaMemcpy }
else if (ROCM_ACCESSIBLE(src) && ROCM_ACCESSIBLE(dst)) { hipMemcpy }
else { unsupported (to add if tests will fail) }

2. copy_from, copy_to to reuse copy_between